_sfp_cage/zyxel.md

```diff
@@ -18,6 +18,32 @@ layout: default
 | SGMII    | ✅                     |
 | Type     | Router                 |
 
+# Flashing a no-brand firmware or firmware downgrade.
+
+1. access via ssh to the router with admin user (admin password is printed on the back of the router)
+2. Disable firmware version and modelcheck by running the following commands
+```
+zycli fwidcheck off
+zycli modelcheck off
+```
+3. You can close the ssh console, do not reboot the router
+4. Open the router web interface and in the maintenance/firmware upgrade section select the "Restore Default Settings After Firmware Upgrade" option.
+5. Select choose file to select the firmware file you want to upload and click Upload.
+6. The router will automatically reboot and should get back up on 192.168.1.1
+
+# FW No-Brand Version V5.70(ACDZ.0)C0 compiled from Zyxel provided OpenSource package.
+The following is a firmware compiled starting from Zyxel provided OpenSource package. 
+
+--- Use at your own risk!!!! ---
+
+- [FW No-Brand Version V5.70(ACDZ.0)C0](https://mega.nz/file/7FZlWTzQ#F8Q_whoW4h1ETRUAzHe4PXIiK6TVj7uay_OtZxfmR6k)
```

Comment:
The firmware is based on sources licensed under the GPL-2.0, GPL-3.0 and other more permissive licenses. Where are the sources of related to this firmware?

Reply:
Zyxel sent a link to me via email with a link to google drive to download the package. The original link only lasts 7 days.

Reply:
You should publish sources for at least GPL-2.0, GPL-3.0 and other programs under copyleft licenses along with the firmware as required by the respective licenses

Reply:
Is it ok a link on my mega to host the original oss package?

Reply:
It would be preferred on Github, Gitlab or similar but Mega is also legit to distribute the sources.